### PR TITLE
Feature/dsci 140 opensearch fields for backend

### PR DESCRIPTION
**Pull request by kdutia**
I've added the following fields to Opensearch documents as per the linear issue:

* `document_date` (created from `published_ts`)
* `document_geography`
* `document_category`
* `document_source`
* `document_type`

I didn't add `publication_ts` directly as it seemed to be referred to as `document_date` in the backend - let me know if that was a mistake.

There's also a small change required to text blocks in the backend: page and text block coords should be optional, as we don't store them for HTMLs. I'll open a PR for that in a minute.

Example Opensearch doc with dummy values:
``` json
{
        "_index" : "navigator",
        "_type" : "_doc",
        "_id" : "_uVcD4QBCf2JoFW5YpaG",
        "_score" : 0.61618614,
        "_source" : {
          "text_block_id" : "b0",
          "text" : "From 1 July 2022, the Department of Industry, Science, Energy and Resources (DISER) becomes the Department of Industry, Science and Resources (DISR). This follows Administrative Arrangements Orders issued on 1 June 2022.",
          "text_embedding" : [...],
          "text_block_coords" : null,
          "text_block_page" : null,
          "document_id" : "test_html",
          "document_name" : "test_html",
          "document_description" : "test_html_description",
          "document_source_url" : "https://www.industry.gov.au/funding-and-incentives/emissions-reduction-fund",
          "document_cdn_object" : null,
          "document_md5_sum" : null,
          "translated" : false,
          "document_slug" : "YYY",
          "document_content_type" : "text/html",
          "document_date" : "25/10/2022",
          "document_geography" : "test_geo",
          "document_category" : "test_category",
          "document_source" : "test_source",
          "document_type" : "test_type"
        }
      },
```

### CODE REVIEW

File: cli/index_data.py [hunk 1]
```diff
@@ -46,7 +46,10 @@ def get_metadata_dict(task: IndexerInput) -> dict:
     :return dict: key-value pairs for metadata fields
     """
 
-    task_dict = {**task.dict(), **task.document_metadata.dict()}
+    task_dict = {
+        **task.dict(),
```
Comment (diversemix) [inline cli/index_data.py:50]: I'm interested - do we really need the metadata in this dict twice? or can we remove the top-level `metadata` key now they're expanded into top-level entries?

File: cli/test/test_index_data.py [hunk 2]
```diff
@@ -30,33 +30,22 @@ def test_get_document_generator(test_input_dir: Path):
...
+            "document_md5_sum",
+            "translated",
+            "document_slug",
+            "document_content_type",
+            "document_geography",
+            "document_category",
+            "document_source",
+            "document_type",
+            "document_date",
+        ]:
```
Comment (diversemix) [inline cli/test/test_index_data.py:47]: should we also check for "metadata" if we're still leaving it in the top-level object?

File: cli/index_data.py (hunk 1 above)
Comment (kdutia) [reply cli/index_data.py]: @diversemix good catch - it's actually filtered out in the next line as `document_metadata` isn't in `src.index_mapping.COMMON_FIELDS` but I've removed it from the top level as it seems more sensible to remove the duplication for any extra work we might do on `task_dict` in the future

File: cli/test/test_index_data.py (hunk 2 above)
Comment (kdutia) [reply cli/test/test_index_data.py:47]: it's no longer in the top-level object - see above comment

**Review by diversemix — APPROVED** [review]
LGTM

**Pull request merged**